tock/hours/models.py

```diff
@@ -108,6 +110,21 @@ def notes_list(self):
         return self.notes.split('\n')
 
     def save(self, *args, **kwargs):
+        """Custom save() method to append employee grade info to each
+        TimecardObject. Grades are only appended if they are in force prior to
+        the end of the reporting period for which the timecard object is being
+        filed. Also updates timecard object with the submitted status of the
+        related timecard."""
+        emp_grd_objects = EmployeeGrade.objects.filter(
+                Q(employee = self.timecard.user)
+                & Q(g_start_date__lte = self.timecard.reporting_period.end_date)
+                ).all().aggregate(Max('g_start_date'))['g_start_date__max']
+
+        if emp_grd_objects:
+            self.grade = EmployeeGrade.objects.filter(
+                employee=self.timecard.user,
+                g_start_date=emp_grd_objects)[0]
+
```

Comment:
Definitely like the logic. Addressed in https://github.com/18F/tock/pull/461/commits/e4698e38888d7844476ee06106b6a4b06e81d5e2 !